OZprivate/rawJS/OZTreeModule/src/position_helper.js

```diff
@@ -244,9 +246,10 @@ function perform_actual_fly(controller, into_node, speed=1, accel_type="linear",
     }
     return true;
   } else {
-    length_intro = Math.abs(Math.log(r_mult))*global_anim_speed;      
-    num_intro_steps = Math.max(Math.floor(length_intro),12)/speed;
-    perform_fly_b2(controller, into_node, speed, accel_type, finalize_func, abrupt_func);
+    fly_duration_s = Math.max(Math.abs(Math.log(r_mult)) * global_anim_speed, 12) / speed / original_flight_fps;
```

Comment:
Whilst it all looks correct to me, I think we have too many magic numbers in this expression at this point. ``global_anim_speed`` as a tweakable value is kinda pointless after this. The point was to fake smooth flights for screencasts, and now we just have smooth flights all the time :)

Does doing something like the following make sense?

```javascript
fly_duration_s = Math.max(Math.abs(Math.log(r_mult)), 12 / global_anim_speed) * global_anim_speed
    / speed / original_flight_fps;
```

... and then ...

```javascript
const original_flight_fps = (1000/60) / 10;  // 10 calibrates frame rate against animation speed values
fly_duration_s = Math.max(Math.abs(Math.log(r_mult)), 1.2)
    / (speed * original_flight_fps);
```

I also think that ``Math.max(x, 1.2)`` is too subtle to be worth it, and only really serves to accentuate the shuffling backwards to the common ancestor when you're basically already on it. My preference would be to bin it, but maybe @jrosindell should weigh in at this point.

Reply:
Yeah I think the units of `global_anim_speed` and `speed` aren't very obvious (some mystery constant factor of 1/scale-doubling time).

That said, I don't think the Math.max is that subtle, the current easing looks quite abrupt at low durations because what you end up with looks like high acceleration.

https://github.com/user-attachments/assets/6c730ab4-3bd8-4496-9425-57f0debbb967

I think I'll leave this as it is tbh, not seeing a clear benefit of reshuffling stuff, certainly if we call it original_flight_fps then it should be in units of frames per second and if we call it something else and stuff the factor of 10 in there then it's kind of just another bizzarely-united value floating around.


https://github.com/user-attachments/assets/501fcb46-9c82-4e65-84f0-6efa3522f0c7